src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -700,6 +705,13 @@ def _prepare_repo(
             logger.error("Cannot reset the target repository.")
             return None
 
+        # Find the digest and branch if a version has been specified
+        if defaults.getboolean("repofinder", "find_commits") and not digest and purl and purl.version:
```

Comment:
Checking if `purl.version` is `None` here is probably not necessary, since you are checking it in the `get_commit_from_version` function anyway.

Reply:
If there is no version we don't want to call the commit finder, because doing so would result in a failed attempt to prepare the repo versus continuing and most likely checking out the latest commit (for a non-versioned PURL),

Reply:
Ah, I see what you mean. The `get_commit_from_version` function should only be called when `purl.version` is present. That makes sense.